src/Tools/DateTimeComparator.php

```diff
@@ -5,57 +5,36 @@
 class DateTimeComparator
 {
     /**
-     * Get the greatest DateTime between the twice given as parameter
-     * If one DateTime is null, return the other
-     * If the twice DateTime are null, return null
+     * Get the greatest "DateTimeInterface" from the parameters
      */
-    public static function greatest(\DateTimeInterface $dateTime1 = null, \DateTimeInterface $dateTime2 = null)
+    public static function greatest(... $datetimes)
     {
-        // - If one DateTime is null, return the other date
-        // - If the twice DateTime are null, return null
-        if (\is_null($dateTime1) || \is_null($dateTime2)) {
-            return self::getNonNullableDateTime($dateTime1, $dateTime2);
+        $greatest = null;
+        foreach ($datetimes as $datetime) {
+            if ($datetime instanceof \DateTimeInterface) {
+                if ($datetime > $greatest || null === $greatest) {
+                    $greatest = $datetime;
+                }
+            }
         }
 
-        return $dateTime1 > $dateTime2 ? $dateTime1 : $dateTime2;
+        return $greatest;
     }
 
     /**
-     * Get the lowest DateTime between the twice given as parameter
-     * If one DateTime is null, return the other
-     * If the twice DateTime are null, return null
+     * Get the lowest "DateTimeInterface" from the parameters
      */
-    public static function lowest(\DateTimeInterface $dateTime1 = null, \DateTimeInterface $dateTime2 = null)
+    public static function lowest(... $datetimes)
     {
-        // - If one DateTime is null, return the other date
-        // - If the twice DateTime are null, return null
-        if (\is_null($dateTime1) || \is_null($dateTime2)) {
-            return self::getNonNullableDateTime($dateTime1, $dateTime2);
+        $lowest = null;
+        foreach ($datetimes as $datetime) {
+            if ($datetime instanceof \DateTimeInterface) {
+                if ($datetime < $lowest || null === $lowest) {
+                    $lowest = $datetime;
+                }
+            }
```

Comment:
It is possible to factorize code with `greatest` function

Reply:
Yep, but IMO it is overkill for now :)